src/charm.py

```diff
@@ -37,6 +38,30 @@
 logger = logging.getLogger(__name__)
 
 
+def evaluate_status(func: Callable) -> Callable:
+    """Decorate `PrometheusJujuExporterCharm` method to perform status evaluation.
+
+    This wrapper can be used to decorate a method (primarily an event handler) from
+    `PrometheusJujuExporterCharm` class. When decorated function is executed, this
+    wrapper will perform final assessment of unit's state and sets unit status.
+    """
+
+    @wraps(func)
+    def wrapper(self: "PrometheusJujuExporterCharm", *args: Any, **kwargs: Any) -> Any:
+        """Execute wrapped method and perform status assessment."""
+        result = func(self, *args, **kwargs)
+
+        exporter_running = self.exporter.running()
+        if isinstance(self.unit.status, ActiveStatus) and not exporter_running:
+            self.unit.status = BlockedStatus("Exporter service is inactive.")
```

Comment:
Perhaps it would be good to also include a suggestion for debugging in the status text. For example: "Exporter service is inactive. Please check service log for more info"

Reply:
In effort to make the message as short as possible, perhaps "Exporter service is inactive. (Check service logs)" ? Or is to direct?

Reply:
I don't see anything wrong with being direct here, so +1 from me.